common/Util-desktop.cpp

```diff
@@ -160,6 +162,56 @@ std::size_t getTotalSystemMemoryKb()
     return totalMemKb;
 }
 
+size_t getMaxConcurrentTCPConnections(const size_t minimum)
+{
+#ifdef __linux__
+    auto readDecimal = [](const char* path, const size_t defaultValue) -> size_t {
+        const int fd = ::open(path, O_RDONLY);
```

Comment:
This is really grim; please read the other functions in that file - and try to be inspired by them. We have a mix of fopen and fgets - and we generally assume that weirdness will not occur there. Writing and reviewing yet another loop like this not necessary for /proc files I believe.